src/lib/operators/table_scan.cpp

```diff
@@ -119,48 +120,63 @@ std::shared_ptr<const Table> TableScan::_on_execute() {
        */
       auto keep_chunk_sort_order = true;
       if (in_table->type() == TableType::References) {
-        auto filtered_pos_lists = std::map<std::shared_ptr<const AbstractPosList>, std::shared_ptr<RowIDPosList>>{};
-
-        for (ColumnID column_id{0u}; column_id < in_table->column_count(); ++column_id) {
-          auto segment_in = chunk_in->get_segment(column_id);
-
-          auto ref_segment_in = std::dynamic_pointer_cast<const ReferenceSegment>(segment_in);
-          DebugAssert(ref_segment_in, "All segments should be of type ReferenceSegment.");
-
-          const auto pos_list_in = ref_segment_in->pos_list();
-
-          const auto table_out = ref_segment_in->referenced_table();
-          const auto column_id_out = ref_segment_in->referenced_column_id();
-
-          auto& filtered_pos_list = filtered_pos_lists[pos_list_in];
-
-          if (!filtered_pos_list) {
-            filtered_pos_list = std::make_shared<RowIDPosList>(matches_out->size());
-            if (pos_list_in->references_single_chunk()) {
-              filtered_pos_list->guarantee_single_chunk();
-            } else {
-              // When segments reference multiple chunks, we do not keep the sort order of the input chunk. The main
-              // reason is that several table scan implementations split the pos lists by chunks (see
-              // AbstractDereferencedColumnTableScanImpl::_scan_reference_segment) and thus shuffle the data. While this
-              // does not affect all scan implementations, we chose the safe and defensive path for now.
-              keep_chunk_sort_order = false;
+        if (matches_out->size() == chunk_in->size()) {
+          // Shortcut - the entire input reference segment matches, so we can simply forward that chunk
+          for (ColumnID column_id{0u}; column_id < in_table->column_count(); ++column_id) {
```

Comment:
Out of interest, we mostly use auto to stick (is this the correct term here?) in for loops (e.g., `auto column_id = ColumnID{0u};`). Shall we slowly try to change that or is there a good reason to not use it here?

Reply:
legacy.